sdk-setup/src/rpmvalidation

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# Copyright (C) 2018 Jolla Ltd.
+# Contact: Ville Nummela <ville.nummela@jolla.com>
+#
+# This file is part of the SailfishOS SDK
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+usage() {
+    cat <<EOF
+Run a basic quality criteria check for the given RPM file.
+
+Usage: 
+    $0 -t <target> [OPTIONS] <rpm-file>
+	
+Options:
+    -t|--target <target> The OS version to validate for. 
+                         Accepts the same argument as 'sb2 -t' does.
+EOF
+}
+
+SUITE_INI_DIR=/usr/share/rpmvalidation
+SUITE_DIR=/usr/libexec/rpmvalidation/suites
+
+validators=()
+
+populate_validators() {
+    VALIDATOR_INIS=$(sb2 -t "$OPT_TARGET" ls "$SUITE_INI_DIR")
+    for f in $VALIDATOR_INIS; do
+        validators+=(${f%.ini})
+    done
+}
+
+validator_value() {
+    sb2 -t "$OPT_TARGET" python3 -c 'import sys, configparser; config = configparser.RawConfigParser(); config.read(sys.argv[2:]); sec, val = sys.argv[1].split("."); print(config[sec][val]);' RpmValidationSuite.$2 /usr/share/rpmvalidation/$1.ini
+}
+
+run_validator() {
+    validator_exec=$(validator_value $1 exec)
+    validator_cmd=$(echo "$validator_exec" | sed -e "s#%f#$RPM_FILE#" -)
+    sb2 -t "$OPT_TARGET" "$validator_cmd"
+}
+
+run_all_validators() {
+    echo "Using validation suites: ${validators[@]}"
+	for f in ${validators[@]}; do
+        run_validator $f
+	done
+}
+
+if [[ $0 == *.sh ]]; then
+    echo "Warning: The name '$0' is deprecated. Use '${0%.sh}' instead"
+fi
+
+if [[ $# -lt 3 ]]; then
+    usage
+    exit 1
+fi
+
+
+while [[ $1 ]]; do
+    case $1 in
+        "-t" | "--target")
+            shift
+            OPT_TARGET=$1
+            ;;
+        *)
+            RPM_FILE=$1
+            ;;
+    esac
+    shift
+done
+
+populate_validators
+
+if [[ $OPT_TARGET ]]; then
```

Comment:
I would handle the error conditions first and implement the normal execution path at the very end of the file. The error messages need a bit more love :)

The script may grow in future - it is a good convention to encapsulate the "main" code to three functions `set_defaults`, `parse_options` and `main`, which allows easier debugging and test writing later (you can e.g `source` (`.`) the script in an interactive shell and debug each function without invoking the main code). Check e.g. sdk-motd for reference.